### PR TITLE
Add protobuf definitions for the ephemeral RPCs

### PR DESCRIPTION
**Pull request by jbardin**
- Add the protobuf definitions for the ephemeral resources, and bump the minor versions to 5.6 and 6.7 to indicate the change.
- The initial POC was missing the validate call, so we need to also add `ValidateEphemeralConfig` throughout the codebase.
- Update all the remaining provider implementations to match the providers interface as well as the generated protobuf interfaces.

### CODE REVIEW

File: internal/builtin/providers/terraform/provider.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func (p *Provider) GetProviderSchema() providers.GetProviderSchemaResponse {
 		ResourceTypes: map[string]providers.Schema{
 			"terraform_data": dataStoreResourceSchema(),
 		},
-		EphemeralResourceTypes: map[string]providers.Schema{},
+		EphemeralTypes: map[string]providers.Schema{},
```
Comment (radeksimko) [inline internal/builtin/providers/terraform/provider.go:36]: Was there a particular motivation behind this rename? It feels to me like we're introducing a third convention here.

I mean, we could have:

1. "data sources" + "resources" + "ephemeral resources"
2. "data source types" + "resource types" + "ephemeral resource types"
3. "data types" + "resource types" + "ephemeral types"

That is to say the current state is _already_ inconsistent but I'm not sure this new name is helping.

File: docs/plugin-protocol/tfplugin6.7.proto [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+    rpc ApplyResourceChange(ApplyResourceChange.Request) returns (ApplyResourceChange.Response);
+    rpc ImportResourceState(ImportResourceState.Request) returns (ImportResourceState.Response);
+    rpc MoveResourceState(MoveResourceState.Request) returns (MoveResourceState.Response);
+    rpc ReadDataSource(ReadDataSource.Request) returns (ReadDataSource.Response);
+
+    //////// Ephemeral Resource Lifecycle
+    rpc ValidateEphemeralConfig(ValidateEphemeralConfig.Request) returns (ValidateEphemeralConfig.Response);
+    rpc OpenEphemeral(OpenEphemeral.Request) returns (OpenEphemeral.Response);
+    rpc RenewEphemeral(RenewEphemeral.Request) returns (RenewEphemeral.Response);
+    rpc CloseEphemeral(CloseEphemeral.Request) returns (CloseEphemeral.Response);
```
Comment (radeksimko) [inline docs/plugin-protocol/tfplugin6.7.proto:298]:
```suggestion
    rpc ValidateEphemeralResourceConfig(ValidateEphemeralResourceConfig.Request) returns (ValidateEphemeralResourceConfig.Response);
    rpc OpenEphemeralResource(OpenEphemeralResource.Request) returns (OpenEphemeralResource.Response);
    rpc RenewEphemeralResource(RenewEphemeralResource.Request) returns (RenewEphemeralResource.Response);
    rpc CloseEphemeralResource(CloseEphemeralResource.Request) returns (CloseEphemeralResource.Response);
```

File: internal/providers/provider.go [hunk 3]
```diff
@@ -108,9 +112,9 @@ type GetProviderSchemaResponse struct {
 	// DataSources maps the data source name to that data source's schema.
 	DataSources map[string]Schema
 
-	// EphemeralResourceTypes maps the name of an ephemeral resource type
+	// EphemeralTypes maps the name of an ephemeral resource type
 	// to its schema.
-	EphemeralResourceTypes map[string]Schema
+	EphemeralTypes map[string]Schema
```
Comment (radeksimko) [inline internal/providers/provider.go:117]: Same question here about the renaming ^

**Review by radeksimko — COMMENTED** [review]
This looks reasonable to me aside from naming. I appreciate this is still a relatively early stage and one PR of many but it would still be great if the DevEx team could start integrating against this without us having to change it later.

Some of the names were also already agreed as part of `TF-949` (protocol) RFC, so I assume @austinvalle used them in his prototypes within the framework etc.

**Comment by radeksimko** [comment]
> Much of the code was using shorter names without Resource, like OpenEphemeralRequest and ValidateEphemeralConfigRequest, while other references were using longer names like OpenEphemeralResourceRequest and ValidateEphemeralResourceConfigRequest. For now I've settled on the shorter version only because it required the least changes to the code. We can do a larger refactor later if the names are not satisfactory.

I have only now read this last part of your PR description - apologies.

I'd still hope that the compiler would make it a bit easier to rename things across the codebase, along with an IDE. That said, if you want to leave it for another PR that's also OK. 

I just wanted to note that such a PR will very likely be _necessary_ and ideally should come relatively soon, so as not to hamper integration.

File: internal/builtin/providers/terraform/provider.go (hunk 1 above)
Comment (jbardin) [reply internal/builtin/providers/terraform/provider.go:36]: It wasn't me 😆 I was just following previously published code.

**Comment by jbardin** [comment]
Yeah, I might as well do all the work here rather than a separate PR. This type of change is always difficult with the language tools, because each rename will break something else somewhere and the tools don't like to do things that render the code invalid.

**Review by radeksimko — APPROVED** [review]
Thank you for making those changes

**Pull request merged**